mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -1986,6 +1986,42 @@ static Value foldScalarExtractFromFromElements(ExtractOp extractOp) {
   return fromElementsOp.getElements()[flatIndex];
 }
 
+// If the dynamic operands of `extractOp` or `insertOp` is result of
+// `constantOp`, then fold it.
+template <typename OpType, typename AdaptorType>
+static Value extractInsertFoldConstantOp(OpType op, AdaptorType adaptor,
+                                         SmallVectorImpl<Value> &operands) {
+  auto staticPosition = op.getStaticPosition().vec();
```

Comment:
Please spell out `auto`. The return type is not obvious:
* https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable